src/clients/HubPoolClient.ts

```diff
@@ -323,7 +381,7 @@ export class HubPoolClient extends BaseAbstractClient {
     latestMainnetBlock: number,
     block: number,
     chain: number,
-    chainIdList: number[]
+    chainIdList = CHAIN_ID_LIST_INDICES
   ): number | undefined {
     let endingBlockNumber: number | undefined;
```

Comment:
We should have the same list stored within the configStoreClient instance. If those arrays are not identical then we'd likely end up with undefined behaviour, so it should be both easier and safer to source the array from there.

```suggestion
  ): number | undefined {
    const chainIdList = this.configStoreClient.enabledChainIds;
    let endingBlockNumber: number | undefined;
```